azure/services/managedclusters/spec.go

```diff
@@ -410,12 +419,6 @@ func (s *ManagedClusterSpec) Parameters(ctx context.Context, existing interface{
 		// AgentPool changes are managed through AMMP.
 		managedCluster.AgentPoolProfiles = existingMC.AgentPoolProfiles
 
-		// Do not trigger an update because of nil/empty discrepancies between the two sets of tags.
```

Comment:
This PR delegates all of the responsibility for diffing and updating `spec.additionalTags` to azure/services/tags, so that kind of logic here is no longer necessary. Basically, we never to trigger an update to a managed cluster resource here based on tags, since azure/services/tags should notice any changes and act accordingly.